src/main/scala/scalaz/stream/Process.scala

```diff
@@ -846,10 +846,10 @@ object Process extends ProcessInstances {
    * Note that the actual granularity of these elapsed times depends on the OS, for instance
    * the OS may only update the current time every ten milliseconds or so.
    */
-  def duration: Process[Task, Duration] = suspend {
-    val t0 = System.nanoTime
-    repeatEval { Task.delay { Duration(System.nanoTime - t0, NANOSECONDS) }}
-  }
+  def duration: Process[Task, Duration] =
```

Comment:
This behaviour to me is weird. Hence suspend shall be lazy and as such I think suspend(f) is kind of equivalent to eval(Task.delay(f)). Do you know frank what are the differences here ?